File: packages/types/src/data-snapshot.ts
export type SnapshotHealthState =
    | "healthy"
    | "degraded"
    | "stale"
    | "failed"
    | "never_synced";

export type TrustLanguageMode = "current" | "qualified" | "historical" | "unavailable";

export interface TrustLanguagePolicy {
    policy_version: number;
    mode: TrustLanguageMode;
    message_key: string;
    health: SnapshotHealthState;
    reason_code: string;
    coverage_status: string;
    pipeline_status: string | null;
    present_tense_allowed: boolean;
    current_trend_allowed: boolean;
    no_alerts_claim_allowed: boolean;
    crisis_claim_allowed: boolean;
    action_mode: "current_if_supported" | "exploratory_only" | "restore_data_first" | "connect_or_restore_data";
    required_qualifier: string | null;
    forbidden_claims: string[];
    next_action: {
        code: "keep_monitoring" | "review_coverage" | "review_partial_run" | "sync_now" | "retry_sync" | "start_first_sync" | "run_analysis";
        href: string;
        priority: "low" | "medium" | "high";
    };
}

export interface SnapshotReference {
    id: string;
    schema_version: number;
    source_platforms: string[];
    profiles: Array<Record<string, unknown>>;
    period_start: string | null;
    period_end: string | null;
    last_attempt_at: string | null;
    last_success_at: string | null;
    found_count: number | null;
    eligible_count: number | null;
    collected_count: number | null;
    saved_count: number | null;
    analyzed_count: number | null;
    valid_count: number | null;
    ignored_count: number | null;
    coverage: Record<string, unknown>;
    health: SnapshotHealthState;
    reason_code: string;
    metrics: Record<string, unknown>;
    content_hash: string;
    created_at: string;
    language_policy: TrustLanguagePolicy;
}

export interface DataSnapshot extends SnapshotReference {
    user_id: string;
    trigger_run_id: string | null;
}

File: backend/app/schemas/data_snapshot.py
import uuid
from datetime import datetime
from typing import Any, Literal

from pydantic import BaseModel, Field


SnapshotHealthState = Literal[
    "healthy",
    "degraded",
    "stale",
    "failed",
    "never_synced",
]


class DataSnapshotResponse(BaseModel):
    id: uuid.UUID
    user_id: uuid.UUID
    trigger_run_id: uuid.UUID | None
    schema_version: int
    period_start: datetime | None
    period_end: datetime | None
    last_attempt_at: datetime | None
    last_success_at: datetime | None
    source_platforms: list[str]
    profiles: list[dict[str, Any]]
    found_count: int | None
    eligible_count: int | None
    collected_count: int | None
    saved_count: int | None
    analyzed_count: int | None
    valid_count: int | None
    ignored_count: int | None
    coverage: dict[str, Any]
    health: SnapshotHealthState
    reason_code: str
    metrics: dict[str, Any]
    content_hash: str = Field(min_length=64, max_length=64)
    created_at: datetime
    language_policy: dict[str, Any]

    model_config = {"from_attributes": True}


class SnapshotReference(BaseModel):
    id: uuid.UUID
    schema_version: int
    source_platforms: list[str]
    profiles: list[dict[str, Any]]
    period_start: datetime | None
    period_end: datetime | None
    last_attempt_at: datetime | None
    last_success_at: datetime | None
    found_count: int | None
    eligible_count: int | None
    collected_count: int | None
    saved_count: int | None
    analyzed_count: int | None
    valid_count: int | None
    ignored_count: int | None
    coverage: dict[str, Any]
    health: SnapshotHealthState
    reason_code: str
    metrics: dict[str, Any]
    content_hash: str
    created_at: datetime
    language_policy: dict[str, Any]

    model_config = {"from_attributes": True}
